### PR TITLE
Port typeclass derivation inline to macros

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Currently blocked by #7405

**Comment by biboudis** [comment]
BTW is `searchImplicit` semantically the same with `summonFrom`? If yes, maybe we can align the terminology. WDYT?

File: tests/run-macros/typeclass-derivation2d/Lib_1.scala [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+  inline def showCase(x: Any, m: Mirror.ProductOf[_]): String = {
+    val label = constValue[m.CaseLabel]
+    inline m match {
+      case m: Mirror.Singleton => label
+      case _ => showElems[m.ElemTypes, m.ElemLabels](0)(x).mkString(s"$label(", ", ", ")")
+    }
+  }
+
+  inline def showCases[Alts <: Tuple](n: Int)(x: Any, ord: Int): String =
+    inline erasedValue[Alts] match {
```
Comment (julienrf) [inline tests/run-macros/typeclass-derivation2d/Lib_1.scala]: What is `erasedValue` necessary?

Comment (nicolasstucki) [reply tests/run-macros/typeclass-derivation2d/Lib_1.scala]: That comes from the original implementation. I have not ported it yet. It is a way to encode a match on a type.

**Comment by nicolasstucki** [comment]
Blocked by #12440

**Pull request closed without merging**